light_node/src/main.rs

```diff
@@ -82,6 +84,8 @@ fn create_logger(env: &crate::configuration::Environment) -> Logger {
 fn block_on_actors(env: &crate::configuration::Environment, identity: Identity, actor_system: ActorSystem, init_info: TezosStorageInitInfo, persistent_storage: PersistentStorage, protocol_commands: IpcCmdServer, protocol_events: IpcEvtServer, protocol_runner_run: Arc<AtomicBool>, log: Logger) {
     let mut tokio_runtime = Runtime::new().expect("Failed to create tokio runtime");
 
+    let ctx_storage = Arc::new(RwLock::new(SkipList::new(0, &persistent_storage).expect("failed to initialize context storage")));
```

Comment:
Also we should hide internal mutability of the skip_list so `RwLock` won't be needed here. It's an implementation detail which the caller of the skip list doesn't need to know of.

Reply:
If we want to use SkipList as a general data structure, this is the way to go. RwLock does not hide inner mutability, just implements the missing thread synchronicity. Only other way would be thread safe rewrite, or some inner RwLock juggling.